### PR TITLE
feat: added unit and integration tests for defer task endpoint

### LINKED ISSUE #55: Feat: Implement `PATCH /v1/tasks/{taskId}?action=defer` API to defer a specific task
### Issue Description

Implement a RESTful API endpoint to allow a user to defer (snooze) a specific task for a specified period. This is a USP feature.

- PRD: https://docs.google.com/document/d/1KOFK8ISE1hy9h8QjNbrObo6QKeM1I-rYZTJ8ygwfxhg/edit?usp=sharing
- Design Doc: https://docs.google.com/document/d/11mhP_qmH7-mnojUUPmtgZR-Vu5Oh-DJltz-ZPzC3jmg/edit?usp=sharing

### Expected Behaviour

When a valid `task_id` and deferral period (e.g., number of days, or a specific `defer_until_date`) are provided:

1. The API should update the task's status or add a deferred_until timestamp.
2. The task should be hidden from the main active task lists until the deferral period ends.
3. The API should return a 200 OK status with the updated task details (showing its new deferred state/date).
4. If the task does not exist, return a 404 Not Found.
5. If the deferral parameters are invalid (e.g., deferring for a negative duration), return a 422 Unprocessable Content.
6. Implement logic as per PRD: "Not allow the user to defer task for further than a certain preset time before the Due Date".

### Current Behaviour

Currently:
1. There is no API endpoint to defer a task.
2. Tasks cannot be snoozed or temporarily hidden.

### Screenshots

- Flow Diagram

Image

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1.  Attempt to `PATCH /v1/tasks/{taskId}?action=defer` with a deferral duration.
2.  Observe that the endpoint does not exist and the task is not deferred.

### Severity/Priority
 - [ ] Critical
 - [x] High (Core USP Feature)
 - [ ] Medium
 - [ ] Low

### Additional Information

- The payload should specify how long to defer the task (e.g., `defer_days: 7` or `defer_until_date: "YYYY-MM-DD"`).
- The task model will need a field to store the `deferred_until` date.
- The "GET /tasks/" endpoint will need to be updated to filter out deferred tasks from the default view (unless a specific filter for deferred tasks is applied).

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by Achintya-Chatterjee**
Date: `June 12, 2025`

Developer Name: @Achintya-Chatterjee 

---

## Issue Ticket Number

- Closes #55 
## Description

- Added tests for defer task endpoint
### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [x] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

Screenshot 2025-06-16 at 18 15 27
</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2025-06-16 at 18 15 27
</details>

### What change is being made?
Add unit and integration tests for the "defer task" endpoint, including validation for date inputs, task state conflicts, and error handling in the deferred task serialization and service logic.

### Why are these changes being made?
These changes ensure robust validation and handling in the "defer task" functionality by covering various edge cases through integration and unit tests, improving overall code reliability. They also improve error messaging consistency by testing exception handlers, aiding in debugging and user feedback in case of incorrect task deferral actions.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: todo/tests/integration/test_task_update_api.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        self.assertEqual(res.status_code, HTTPStatus.BAD_REQUEST)
+
+        body = res.json()
+        self.assertEqual(body["statusCode"], HTTPStatus.BAD_REQUEST)
+        self.assertEqual(body["message"], ValidationErrors.INVALID_TASK_ID_FORMAT)
+
+        err = body["errors"][0]
+        self.assertEqual(err["title"], ApiErrors.VALIDATION_ERROR)
+        self.assertEqual(err["detail"], ValidationErrors.INVALID_TASK_ID_FORMAT)
+        self.assertEqual(err["source"]["path"], "task_id")
```
[the remark this answers is not in the document]
Comment (Achintya-Chatterjee) [reply todo/tests/integration/test_task_update_api.py:90]: Created, kindly see the  #88

**Review by iamitprakash — APPROVED** [review]
